### PR TITLE
Add benchmark and vignette

### PR DESCRIPTION
**Pull request by RasmusSkytte**
### Intent
This PR adapts the benchmarks created for `SCDB` to `diseasystore`.

Both the `$get_feature()` and `$key_join_features()` functions are benchmarked across different back-ends
and SCDB versions.

### Approach
Benchmarks taken from `SCDB` and adapted.
(To see the changes to this file in particular, see this [link](https://github.com/ssi-dk/diseasystore/pull/144/files/d2009ace33eb42c53f6ee00081e558ba87e424f5..HEAD#diff-eb37369b4a58e2a1164c5b2e4bd70e32bab820849c61e854e71266b2c6d1ce72))

As you can see in the link, the benchmark script has been updated from the SCDB version.
Here there are many differences from the original, which aims to solve the following:
* Benchmark not just against the version of `diseasystore` but also against the version of `SCDB`.
  This way, we can both see how updates to `diseasystore` changes the benchmarks, but also how 
  changes to `SCDB` changes the benchmarks.

	* This means we need to install both versions of `SCDB` and change how the looping is done during benchmarks.

* We need a different thing to benchmark than SCDB (which benchmarked `update_snapshot`).
  We here need to benchmark `get_feature()` and `key_join_features()`. So we build a dummy diseasystore
  that uses mtcars as data, and has some simple features computed from this data.

The workflow (`benchmark.yaml`) is essentially unchanged from the `SCDB` version but now lives on AEF-DDF repository as a reusable workflow. This PR now uses this workflow.

### Known issues
~~This PR builds on un-merged PRs~~
* [x] ~~#172~~

### Checklist

* [x] The PR passes all local unit tests
* [x] I have documented any new features introduced
* [x] If the PR adds a new feature, please add an entry in `NEWS.md`
* [x] A reviewer is assigned to this PR

### CODE REVIEW

**Comment by RasmusSkytte** [comment]
This PR has been stale for a while, so I have rebased onto main and I am giving it a fresh look

**Comment by RasmusSkytte** [comment]
@SofiaOtero 

Since we went through the PR, I have fixed a bug and made some changes to the figure.
There was an issue where the first iteration would take longer than the others (due to having extra overhead such as creating the log table). So now, we do one iteration and clean up before starting the benchmark loop.

billede

File: vignettes/benchmarks.Rmd [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+variables.
+```{r, eval = FALSE}
+DiseasystoreMtcars <- R6::R6Class(
+  classname = "DiseasystoreBase",
+  inherit = DiseasystoreBase,
+  private = list(
+    .ds_map = list("n_cyl" = "mtcars_cyl", "vs" = "mtcars_vs"),
+    mtcars_cyl = FeatureHandler$new(
+      compute = function(start_date, end_date, slice_ts, source_conn) {
+        out <- benchmark_data |>
```
Comment (SofiaOtero) [inline vignettes/benchmarks.Rmd:32]: Would be good to see how `benchmark_data` is build/where it comes from

**Review by SofiaOtero — APPROVED** [review]
Nice with the changes to the vignette, it is much clearer now with the plots and describing text. Only add on would be the comment i made above.

File: vignettes/benchmarks.Rmd (hunk 1 above)
Comment (RasmusSkytte) [reply vignettes/benchmarks.Rmd:58]: Done!

**Pull request merged**